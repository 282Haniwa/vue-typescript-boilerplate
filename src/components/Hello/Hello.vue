<template>
  <h1 class="hello--red">Hello {{ name }}!</h1>
</template>

<script lang="ts">
import Vue from 'vue';

const options = {
  props: {
    name: {
      type: String,
      default: '',
      required: false,
    },
  },
};

export default Vue.extend(options);
</script>

<style lang="scss" scoped>
/* @import "src/assets/sass/global"; */
.hello {
  &--red {
    color: $color-red;
  }
}
</style>
