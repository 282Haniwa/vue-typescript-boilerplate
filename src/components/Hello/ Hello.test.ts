import { shallowMount } from '@vue/test-utils';
import Hello from './Hello.vue';

describe('Hello.vue', () => {
  test('Hello component test', () => {
    const name = 'World';
    const wrapper = shallowMount(Hello, {
      propsData: { name },
    });
    expect(wrapper.text()).toMatch(`Hello ${name}!`);
    expect(wrapper).toMatchSnapshot();
  });
});
