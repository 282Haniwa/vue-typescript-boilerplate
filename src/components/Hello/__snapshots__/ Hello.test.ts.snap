// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello.vue Hello component test 1`] = `<h1 class="hello--red">Hello World!</h1>`;
